Adding information about launching on iOS Device from menu (#678)
spatialos/gdk-for-unity: docs/content/mobile/ios/local-deploy.md

[//]: # (TODO - get rid of mobile_launch.json mention and explain it differently)

> # Connect to a local deployment

Reviewer: best to revert those changes. @oblm is redoing them in his branch afaik

Author: Yup, I'll rebase on top of his changes once they have landed

>   * [Setting up iOS Support for the GDK]({{urlRoot}}/content/mobile/ios/setup)

Reviewer 2: Page title
```suggestion
# Connect  to a local deployment
```

  * [Ways to test your iOS client]({{urlRoot}}/content/mobile/ios/ways-to-test)

> 

Reviewer 2: ```suggestion
There are three different ways to connect your iOS device to a local deployment, however, before you connect, you need to first prepare your project.
```

## Prepare your project to connect to a local deployment

> 

Reviewer 2: ```suggestion
Follow the steps below to connect your iOS device or simulator to a local deployment. The steps ensure that any local deployment you launch via the Unity Editor connects to your iOS device or simulator correctly.
```

> To connect your iOS device or simulator to a local deployment, follow these steps:

Reviewer 2: ```suggestion

```
Merged into the sentence above.

1. Open your project in the Unity Editor.
1. Navigate to **SpatialOS** > **GDK Tools configuration** to open the configuration window.
1. In the **Runtime IP for local deployment** field, enter your local machine's IP address. (You can find how to do this on the [Lifehacker website](https://lifehacker.com/5833108/how-to-find-your-local-and-external-ip-address).)
1. Select **Save**, and close the window.

> This ensures that any local deployment that is launched via the Unity Editor is set up correctly to connect to your iOS device or simulator.

Reviewer 2: ```suggestion

```
Moved to the top of the section.

> 

Reviewer 2: ```suggestion
## Connect to a local deployment
There are three different ways to connect.
```

> ## Connecting your iOS device to a local deployment using Unity Remote

Reviewer 2: ```suggestion
### Connect your iOS mobile device using Unity Remote
```

    >
    > For subsequent runs you will be prompted to pick XCode project directory again (with the one used previously preselected). You can generate a new project or append/replace existing one. In subsequent runs, if you've set up provisioning and use Append, you can use **Build and Run** to trigger project run automatically after XCode project was generated.

  1. Once the game is running on your device, you need to connect using the local IP address.<br/>Either: 
  
      * On your device, note the empty text field and a **Connect** button: enter the local IP address of your computer in the text field and click **Connect**.
      
      Or:
  

>       * In your Unity Editor, navigate to: **SpatialOS** > **Launch Mobile Client** > **iOS Client** and launch mobile client with IP address field prefilled. (It's prefilled with a value from **SpatialOS** > **GDK Tools Configuration** > **Runtime IP for local deployment**.)

Reviewer: ```suggestion
      * In your Unity Editor, navigate to: **SpatialOS** > **Launch Mobile Client** > **iOS Client** to launch the iOS client with the IP address field prefilled. (It's prefilled with the value you entered in the GDK Tools Configuration Window. You can access this by navigating to **SpatialOS** > **GDK Tools Configuration** > **Runtime IP for local deployment** in the Unity Editor.)
```

>       > To use this workflow you need to have [ideviceinstaller](https://github.com/libimobiledevice/ideviceinstaller) and [idevicedebug](https://helpmanual.io/help/idevicedebug/) tools installed.

Reviewer: this should be in the setup guide

Author: Fair enough. I'll add that to set up, but I would also prefer to keep it here - those tools aren't necessary to work, so people might skip this initially.

      >

>       > Make sure .ipa archive is built to `workers\unity\build\` directory.

Reviewer: ```suggestion
      > To start the iOS client from the Unity Editor, you need to built the application to the `workers\unity\build` directory.
```

Author: I would want us to mention that it has to be .ipa archive. There are a few ways that can be considered "building the app", and not all of them produce .ipa that we need.

>       > Make sure .ipa archive is built to `workers\unity\build\` directory.

Reviewer 2: ```suggestion
      > To start the iOS client from the Unity Editor, you need to build the application as a `.ipa` file to the `workers\unity\build` directory.
```
Put both in. Also note we need "build" present tense.

    > For subsequent runs you will be prompted to pick XCode project directory again (with the one used previously preselected). You can generate a new project or append/replace existing one. In subsequent runs, if you've set up provisioning and use Append, you can use **Build and Run** to trigger project run automatically after XCode project was generated.

>   1. Once the game is running on your device, you see an empty text field and a **Connect** button: enter the local IP address of your computer in the text field and click **Connect**.

Reviewer 2: The numbering isn't right on the page when it renders in GitHub - wondering if it's right in improbadoc?
(In the directory above the docs enter `improbadoc serve docs` to get a the HTML docs locally and take a look.)

>   1. Once the game is running on your device, you see an empty text field and a **Connect** button: enter the local IP address of your computer in the text field and click **Connect**.

Reviewer 2: ```suggestion
  1. Once the game is running on your device, you need to connect using the local IP address.<br/>
  Either: 
  
  * On your device, note the empty text field and a **Connect** button: enter the local IP address of your computer in the text field and click **Connect**.
  
  Or: 
```

>   1. Alternatively you can use **SpatialOS** > **Launch Mobile Client** > **iOS Client** to launch mobile client with IP address field prefilled with a value from **SpatialOS** > **GDK Tools Configuration** > **Runtime IP for local deployment**.

Reviewer: take a look at Martijn's PR for adb. We want to get rid of the old way and only use the menu items as that is how we do it for the other workers

Author: Chatted about this offline. Building everything in one go is quite lengthy, so we'll do it in a separate menu item - UTY-1600

>   1. Alternatively you can use **SpatialOS** > **Launch Mobile Client** > **iOS Client** to launch mobile client with IP address field prefilled with a value from **SpatialOS** > **GDK Tools Configuration** > **Runtime IP for local deployment**.

Reviewer 2: 1. As this is an alternative to the step above, it needs to be the same numbered step. 
So, I suggest formatting the two steps as list items.
2. Where do I go to for **SpatialOS** > **Launch Mobile Client** > **iOS Client**  ? Is it in the Unity Editor? If so, we need to say that because we were just talking about the mobile device.
3. Where

>   1. Alternatively you can use **SpatialOS** > **Launch Mobile Client** > **iOS Client** to launch mobile client with IP address field prefilled with a value from **SpatialOS** > **GDK Tools Configuration** > **Runtime IP for local deployment**.

Reviewer 2: ```suggestion
  * In your Unity Editor, navigate to: **SpatialOS** > **Launch Mobile Client** > **iOS Client** and launch mobile client with IP address field prefilled. (It's prefilled with a value from **SpatialOS** > **GDK Tools Configuration** > **Runtime IP for local deployment**.)
```

>     > To use this flow you need to have [ideviceinstaller](https://github.com/libimobiledevice/ideviceinstaller) and [idevicedebug](https://helpmanual.io/help/idevicedebug/) tools installed.

Reviewer 2: ```suggestion
    > To use this workflow you need to have [ideviceinstaller](https://github.com/libimobiledevice/ideviceinstaller) and [idevicedebug](https://helpmanual.io/help/idevicedebug/) tools installed.
```
"workflow" seems nicer but it's not vital to change.

    >

>     > Make sure .ipa archive is built to `workers\unity\build\` directory.

Reviewer 2: 1. Who sets up where the .ipa archive builds, our user?
2. How do they do this?
3. Needs "the" for the archive.
4. Formatting tip: '.ipa'  is in backticks as it's a type of file extension.

```suggestion
    > Make sure the `.ipa` archive builds to the `workers\unity\build\` directory. You can set this <how to set it>.
```

  1. Once the game is running on your device, you need to connect using the local IP address.<br/>Either: 
  
      * On your device, note the empty text field and a **Connect** button: enter the local IP address of your computer in the text field and click **Connect**.
      
      Or:
  
      * In your Unity Editor, navigate to: **SpatialOS** > **Launch Mobile Client** > **iOS Client** to launch the iOS client with the IP address field prefilled. (It's prefilled with the value you entered in the GDK Tools Configuration Window. You can access this by navigating to **SpatialOS** > **GDK Tools Configuration** > **Runtime IP for local deployment** in the Unity Editor.)
  
      > To use this workflow you need to have [ideviceinstaller](https://github.com/libimobiledevice/ideviceinstaller) and [idevicedebug](https://helpmanual.io/help/idevicedebug/) tools installed.
      >
      > To start the iOS client from the Unity Editor, you need to build the application as a `.ipa` archive to the `workers\unity\build` directory.

>   1. Play the game on your mobile device.

Reviewer 2: Suggest slight rejig of the titles.

```suggestion
### Connect your iOS simulator
```

```suggestion
### Connect your iOS mobile device
```

Also, should we have the two mobile device options next to each other, with the simulator last or at the top?
